### ISSUE TITLE
[BUG] Spurious warning about `gluonts` data containers when none is used

### ISSUE THREAD

**Issue by kartik-555** [sktime/sktime#7746]
**Describe the bug**

Getting warning to install 'orjson', 'ujson' while running a script that uses sktime and gluonts.

**To Reproduce**

```python
from sktime.clustering.k_means import TimeSeriesKMeansTslearn
from sklearn.datasets import make_blobs
X_train, _ = make_blobs(n_samples=100, centers=3, n_features=10, random_state=42)
model = TimeSeriesKMeansTslearn(n_clusters=3, metric="euclidean")
model.fit(X_train)
y_pred = model.predict(X_train)
print(y_pred)
```

**Expected behavior**

No warning message should be displayed, and the script should run without suggesting additional installations.

**Additional context**
 
It says "/sktime/env/lib/python3.12/site-packages/gluonts/json.py:102: UserWarning: Using `json`-module for json-handling. Consider installing one of `orjson`, `ujson` to speed up serialization and deserialization.
  warnings.warn(
"

**Versions**
<details>

System:
    python: 3.12.3 (main, Jan 17 2025, 18:03:48) [GCC 13.3.0]
executable: /home/kartik/Desktop/sktime/env/bin/python
   machine: Linux-6.8.0-52-generic-x86_64-with-glibc2.39

Python dependencies:
          pip: 25.0
       sktime: 0.35.0
      sklearn: 1.5.2
       skbase: 0.12.0
        numpy: 1.26.4
        scipy: 1.15.1
       pandas: 2.2.3
   matplotlib: 3.10.0
       joblib: 1.4.2
        numba: 0.60.0
  statsmodels: 0.14.4
     pmdarima: 2.0.4
statsforecast: 2.0.0
      tsfresh: None
      tslearn: 0.6.3
        torch: None
   tensorflow: 2.18.0

</details>

**Comment by kartik-555** [comment]
The estimator should have no dependencies, and I cannot see why json would be used.

**Comment by fkiraly** [comment]
I can reproduce the warning on windows, python 3.11.

The warning happens if `gluonts` is installed, if and only then.

Is in an environment that has `sktime` minimal deps, `tslearn`, but not `gluonts`, the warning will not be shown.

**Comment by fkiraly** [comment]
I understood the deeper mechanism - the originally intended fix in #7736 would have fixed it, but was masked by a bug as some checkers were calling the private instead of the public method.

Fixing the dispatch to the public method - and some secondary bugs (this also caused missing coverage of the boilerplate) - has fixed the issue now, see https://github.com/sktime/sktime/pull/7748
